pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -9484,5 +9485,164 @@ spec:
 
 	verifyTaskRunStatusesCount(t, embeddedStatus, reconciledRun.Status, 2)
 	verifyTaskRunStatusesNames(t, embeddedStatus, reconciledRun.Status, tr1Name, tr2Name)
+}
+
+func TestReconcile_CreateTaskRunWithComputeResources(t *testing.T) {
+	multiStepsTask := parse.MustParseTask(t, `
```

Comment:
really minor: from reading the test, it seems like we only need a very basic task. Instead of a separate task could we just embed it within the pipeline i.e using a `taskSpec` instead of a `taskRef` to refer to this pipeline

Reply:
nice hint. thanks Dibyo!